File: source_funcs/go_transcoder/SWAP_BITS_IN_A_GIVEN_NUMBER/SWAP_BITS_IN_A_GIVEN_NUMBER_emit.go
package main

//go:export f_gold
func f_gold(x uint, p1 uint, p2 uint, n uint) int {
	var (
		set1 uint = (x >> p1) & ((1 << n) - 1)
		set2 uint = (x >> p2) & ((1 << n) - 1)
		Xor  uint = (set1 ^ set2)
	)
	Xor = (Xor << p1) | Xor<<p2
	var result uint = x ^ Xor
	return int(result)
}

File: rust_funcs/cpp_transcoder/SWAP_BITS_IN_A_GIVEN_NUMBER/SWAP_BITS_IN_A_GIVEN_NUMBER_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


#[no_mangle]
fn f_gold(x: u32, p1: u32, p2: u32, n: u32) -> i32 {
    let mut set1 = (x >> p1) & ((1u32 << n) - 1);
    let mut set2 = (x >> p2) & ((1u32 << n) - 1);
    let mut xor = (set1 ^ set2);
    xor = (xor << p1) | (xor << p2);
    let result = x ^ xor;
    result 
}
